auth/src/main/java/org/aerogear/auth/IUserPrincipal.java

```diff
@@ -0,0 +1,30 @@
+package org.aerogear.auth;
+
+import org.aerogear.auth.credentials.TokenCredentials;
+
+import java.security.Principal;
+
+/**
+ * Public interface for user principals.
+ */
+public interface IUserPrincipal extends Principal {
+
+    /**
+     * Checks if the user has the specified role.
+     * @param role role to be checked
+     * @return true or false
+     */
+    boolean hasRole(IRole role);
+
+    /**
+     * Returns the username
+     * @return the username
+     */
+    String getName();
+
+    /**
+     * Returns the session token.
+     * @return the session token
+     */
+    TokenCredentials getToken();
+}
```

Comment:
I think it might be a good idea to expose some more OpenID specific user attributes:

`getName()` - the user's IDM given name.
`getEmail()` - the user's IDM given email

Also maybe a function per token type - 

`getAccessToken()` - Returns the users access token
`getIdentityToken()` - Returns the users identity token
`getRefreshToken()` - Returns the users refresh token